From ee0f179cbad40b0235b07e3c0ed017878cac6d3c Mon Sep 17 00:00:00 2001
From: "/setup.sh" <support@vision-components.com>
Date: Wed, 4 Dec 2024 17:34:14 +0100
Subject: [PATCH 19/19] The function tegracam_v4l2subdev_register did not load
 when the min_gain_val or min_exp_time has been set to 0 in the device tree.

---
 nvidia/drivers/media/platform/tegra/camera/tegracam_ctrls.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/nvidia/drivers/media/platform/tegra/camera/tegracam_ctrls.c b/nvidia/drivers/media/platform/tegra/camera/tegracam_ctrls.c
index 4c5770e81793..0957f6e18c1e 100644
--- a/nvidia/drivers/media/platform/tegra/camera/tegracam_ctrls.c
+++ b/nvidia/drivers/media/platform/tegra/camera/tegracam_ctrls.c
@@ -687,7 +687,7 @@ int tegracam_init_ctrl_ranges_by_mode(
 		switch (ctrl->id) {
 		case TEGRA_CAMERA_CID_GAIN:
 			err = v4l2_ctrl_modify_range(ctrl,
-				ctrlprops->min_gain_val - 1,
+				ctrlprops->min_gain_val,
 				ctrlprops->max_gain_val,
 				ctrlprops->step_gain_val,
 				ctrlprops->default_gain);
@@ -701,7 +701,7 @@ int tegracam_init_ctrl_ranges_by_mode(
 			break;
 		case TEGRA_CAMERA_CID_EXPOSURE:
 			err = v4l2_ctrl_modify_range(ctrl,
-				ctrlprops->min_exp_time.val - 1,
+				ctrlprops->min_exp_time.val,
 				ctrlprops->max_exp_time.val,
 				ctrlprops->step_exp_time.val,
 				ctrlprops->default_exp_time.val);
-- 
2.43.0

